nemo/collections/asr/losses/pt_losses/contrastive.py

```diff
@@ -0,0 +1,206 @@
+# ! /usr/bin/python
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from nemo.core import Loss, typecheck
+from nemo.core.neural_types import LossType, NeuralType, SpectrogramType, VoidType
+
+__all__ = ["ContrastiveLoss"]
+
+
+def buffered_arange(max):
+    if not hasattr(buffered_arange, "buf"):
```

Comment:
hasattr and assignment on a function?? make it a class with `__call__` please, sure it works but syntactically awkward